x-pack/plugins/triggers_actions_ui/public/application/sections/alert_summary_widget/components/alert_counts.tsx

```diff
@@ -14,32 +14,61 @@ import { AllAlertCounts } from './all_alert_counts';
 interface Props {
   activeAlertCount: number;
   recoveredAlertCount: number;
-  onActiveClick?: (
+  handleClick?: (
     event: MouseEvent<HTMLAnchorElement | HTMLDivElement>,
     status?: AlertStatus
   ) => void;
 }
 
-export const AlertCounts = ({ activeAlertCount, recoveredAlertCount, onActiveClick }: Props) => {
+/** @internal **/
+const AlertItem = ({
```

Comment:
nit: What about moving this logic to a separate file and then using it directly in `AllAlertCounts` and `ActiveAlertCounts`?

In that case, we can also move `onAllClick` and `onActiveClick` to the respective components.
In the current implementation, we have some parts of the let's and activeAlerts logic here and some parts of it in the children components. So, in the end, the component here will look something like:
```
<EuiFlexGroup gutterSize="l" responsive={false}>
    <AllAlertCounts count={activeAlertCount + recoveredAlertCount} onClick={onAllClick} data-test-subj="allAlerts" />
    <ActiveAlertCounts activeAlertCount={activeAlertCount} onClick={onActiveClick} data-test-subj="activeAlerts" />
</EuiFlexGroup>
```

Reply:
@maryam-saeidi, could you please review the latest changes? I'm unsure if we need both `AllAlertCounts` and `ActiveAlertCounts`. I've decided to retain only `AlertItem` to remove duplication.